core/src/main/java/org/commonjava/indy/core/expire/ScheduleManager.java

```diff
@@ -251,8 +263,14 @@ public void setProxyTimeouts( final StoreKey key, final String path )
         }
 
         int timeout = config.getPassthroughTimeoutSeconds();
-        final ConcreteResource resource = new ConcreteResource( LocationUtils.toLocation( repo ), path );
-        final SpecialPathInfo info = specialPathManager.getSpecialPathInfo( resource, key.getPackageType() );
+        ConcreteResource resource = new ConcreteResource( LocationUtils.toLocation( repo ), path );
+        ConcreteResource specialResource = resource;
+        for ( final StoragePathCalculator calculator : pathCalculators )
+        {
+            final String pathGened = calculator.calculateStoragePath( key, path );
+            specialResource = new ConcreteResource( LocationUtils.toLocation( repo ), pathGened );
+        }
+        final SpecialPathInfo info = specialPathManager.getSpecialPathInfo( specialResource, key.getPackageType() );
```

Comment:
I think I'd rather see SpecialPathManager inject the CacheProvider, then try a two-level approach to finding SpecialPathInfo:

1. Look at the logical path in `ConcreteResource.getPath()` to see if there's special info about that
2. If not, use `CacheProvider.getStoragePath(resource)` and look to see if that path has special info about it.

This would make the logic more self-contained for things that care about special paths, so we don't have to replicate this all over the place.

Reply:
Agree :+1: I've did this change in https://github.com/Commonjava/galley/pull/285, and change indy to use it in new commit.